### ISSUE TITLE
cyclocomp linter is not actionable

### ISSUE THREAD

**Issue by mcol** [r-lib/lintr#2943]
I received this message but I don't have a clue on what I should be doing:
```
style: [cyclocomp_linter] Reduce the cyclomatic complexity of this function from 32 to at most 15.
```
Perhaps just some generic advice should be reported alongside that message. Even better would be to lint the places that are considered highly complex and thus would help reduce the score.

**Comment by MichaelChirico** [comment]
Yea, I agree it's not the most actionable. I wonder how helpful it would be to do a leave-one-out computation like:

`for (j in seq_along(body(FUN))) { cylcocomp(exclude_expr(FUN, jj)) }` to at least point to the top-level expression that adds the most to the complexity.

One issue is that `cyclocomp_linter()` is already one of the slowest in the suite, this would substantially increase the cost.

The most generic advice would be something like:

> Consider replacing high-complexity sections like loops and branches with helper functions.

**Comment by MichaelChirico** [comment]
I think probably the best solution is to suggest the user run a {cyclocomp} diagnostic themselves to understand how to proceed... but there is no such helper today!

In fact, it turns out I already made this FR 2 years ago :) https://github.com/gaborcsardi/cyclocomp/issues/24
